nessie/store_test.go

```diff
@@ -0,0 +1,35 @@
+package nessie
+
+import (
+	"context"
+	"net/url"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/stretchr/testify/require"
+	"github.com/treeverse/lakefs/cmd/lakectl/cmd/store"
+)
+
+const IngestTestBacketPath = "s3://nessie-system-testing-data/ingest-test-data/"
+
+func TestS3Walk(t *testing.T) {
```

Comment:
Using the Walk function, we can test all the storage we support - not saying we need to add configuration currently to Nessie now. But writing the test function using table-driven style, we add just pass different bucket Path
Can drop the S3 from the name and use table-driven code, so adding another address we can check more storage systems.

Something like:
```go
func TestWalk(t *testing.T) {
	// Test bucket was uploaded with 2100 as the test is written. If the test fails on this number,
	// make sure there were no changes made to the bucket, or update this number accordingly
	const expectedNumObjs = 2100
	tests := []struct {
		Name       string
		BucketPath string
	}{
		{Name: "s3", BucketPath: "s3://nessie-system-testing-data/ingest-test-data/"},
	}

	for _, tt := range tests {
		t.Run(tt.Name, func(t *testing.T) {
			uri, err := url.Parse(tt.BucketPath)
			require.NoError(t, err, "URL parsing failed", tt.BucketPath)
			numObjs := 0
			store.Walk(context.Background(), uri, func(store.ObjectStoreEntry) error {
				numObjs++
				return nil
			})
			require.Equal(t, expectedNumObjs, numObjs, "Wrong number of objects detected", tt.Name)
		})
	}
}

```

Reply:
I can open an issue to address this in the future - WDYT?

Reply:
Opened #3101 and #3102

Reply:
Think you can close/delete 3102, we would like to run all the tests and locally with the suggested code we can run specific storage.